docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -942,6 +943,56 @@ const (
 )
 ```
 
+Here is the example of runtime with gang-scheduling using coscheduling plugin.
+
+```yaml
+apiVersion: kubeflow.org/v2alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: torch-distributed-multi-node
+spec:
+  mlSpec:
+    torch:
+      numProcPerNode: 5
+  podGroupSpec:
+    plugin: coscheduling
+    scheduleTimeoutSeconds: 100
+  replicatedJobs:
+    - name: node
+      template:
+        spec:
+          template:
+            spec:
+              schedulerName: coscheduling
+              containers:
+                - name: trainer
+                  image: docker.io/kubeflow/pytorch-mnist
+                  resources:
+                    limits:
+                      nvidia.com/gpu: 1
+                  env:
+                    - name: MASTER_ADDR
+                      value: "pytorch-node-0-0.pytorch"
+                    - name: MASTER_PORT
+                      value: 29400
+                  command:
+                    - torchrun train.py
+```
+
+Training Operator will create the `PodGroup` using the following spec:
```

Comment:
This makes sense. Naive question - does this imply that any custom gang scheduler used is required to make a PodGroup to function? Or if you have a plugin that doesn't create a PodGroup (and does its own thing) the group would be made anyway? And from the example above, how is `minMember` of 5 derived? You would probably need custom logic per scheduler backend for this API. For example, the PodGroup created by coscheduling vs volcano.sh are different underlying objects. (https://github.com/kubernetes-sigs/scheduler-plugins/blob/d67d2c15199595ccc6218574bd8e07b68b7146f4/apis/scheduling/v1alpha1/types.go#L128  and https://github.com/volcano-sh/apis/blob/78d912ce096c9fd9120488b5c7b999d4996f2cb5/pkg/apis/scheduling/types.go#L147)

Reply:
If this Training API is going t take on the challenge of orchestrating the logic for creation of needed PodGroup (varying by the plugin) then this probably works OK. Otherwise, I'd still place the responsibility on the user for now for creating the pod group, but just allow the `schedulerName` and other labels to use it in whatever turns into the actual job or podspec.

Reply:
>does this imply that any custom gang scheduler used is required to make a PodGroup to function? 

The PodGroup creation will be implemented only for supported gang-schedulers (initially Volcano and Coscheduling). Since different schedulers require [the various PodGroups to be created](https://github.com/kubeflow/training-operator/blob/master/pkg/controller.v1/common/job.go#L281).

> And from the example above, how is minMember of 5 derived? 

`minMember` is always equal to `numNodes`. For ML Training all gangs should be alive to execute training. In the future, if we find other use-cases (e.g. HPC), we can discuss it again.

>If this Training API is going t take on the challenge of orchestrating the logic for creation of needed PodGroup (varying by the plugin) then this probably works OK

Yeah, that's correct.

>Otherwise, I'd still place the responsibility on the user for now for creating the pod group, but just allow the schedulerName and other labels to use it in whatever turns into the actual job or podspec.

Btw, this also will be supported, since users can just ignore `PodGroupSpec` parameter, and set the `.spec.schedulerName` field in the PodSpec.